centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala

```diff
@@ -107,7 +107,25 @@ object CentaurCromwellClient extends StrictLogging {
     Try(Await.result(successOrFailure, CentaurConfig.sendReceiveTimeout)).isSuccess
   }
 
-  def metadata(workflow: SubmittedWorkflow, args: Option[Map[String, List[String]]] = defaultMetadataArgs): IO[WorkflowMetadata] = metadataWithId(workflow.id, args)
+  def metadata(workflow: SubmittedWorkflow,
+               args: Option[Map[String, List[String]]] = defaultMetadataArgs,
+               expandSubworkflows: Boolean = false,
+               archived: Option[Boolean] = None): IO[WorkflowMetadata] = {
+    val mandatoryArgs = Map("expandSubWorkflows" -> List(expandSubworkflows.toString))
+
+    val metadataSourceOverrideOpt = archived map { isArchived =>
+      "metadataSource" -> List(if (isArchived) "Archived" else "Unarchived")
+    }
+    val optArgs = args match {
+      case None => metadataSourceOverrideOpt.map(Map(_))
+      case Some(argMap) =>
+        metadataSourceOverrideOpt match {
+          case Some(metadataSourceOverride) => Option(argMap + metadataSourceOverride)
+          case None => Option(argMap)
+        }
+    }
+    metadataWithId(workflow.id, Option(mandatoryArgs ++ optArgs.getOrElse(Map.empty)))
```

Comment:
You can use `metadataSourceOverrideOpt.toMap` to get a map of either 0 or 1 elements, which should make this composition a little easier:
```
mandatoryArgs ++ 
  metadataSourceOverrideOpt.toMap ++
  args.getOrElse(Map.empty)
```